app/javascript/src/components/dashboard/Hours.jsx

```diff
@@ -0,0 +1,242 @@
+  customCard = (props) => {
+  return <Card key={props.user.name} style={{
+        width: "180px"
+      }}>
+      <Card.Content >
+        <Card.Header>{props.user.name}
+        </Card.Header>
+      </Card.Content>
+      <Card.Content>
+        {this.state.total == false ? "Week " : "Total "}
+        <Card.Description>
+          {this.hourView()}
+          <strong>{this.hoursNumber(props.user)}
+          </strong>
+        </Card.Description>
+        <Card.Description>
+          {this.nightView()}
+          <strong>{this.nightNumber(props.user)}
+          </strong>
+        </Card.Description>
+      </Card.Content>
+      <Card.Content >
+        <Grid.Column>
+          Hours Rank: {this.hourRank(props.user)}
+          {this.trophy(props.user, "hour")}
+        </Grid.Column>
+        <Grid.Column>
+          Nights Rank: {this.nightRank(props.user)}
+          {this.trophy(props.user, "night")}
+        </Grid.Column>
+      </Card.Content>
+    </Card>
+  }
+
+  //Needed to updated 'total' state value while switching view in dropdown. Look at line 192
+  handleChange(e, timeofDay) {
+    const total = timeofDay === 'Week'
+      ? false
+      : true;
+    this.setState({
+      total
+    }, () => {
+      this.handleNightorHours(e, this.state.rank)
+    });
+  };
+
+  // Trying to move the dropDown options out of the render component, but dropDown fucntion  is buggy
+
+  dropDown = () => {
+    const lengthOptions = [
+      {
+        key: 'Week',
+        text: 'Week',
+        value: 'Week',
+        content: 'Week'
+      }, {
+        key: 'Total',
+        text: 'Total',
+        value: 'Total',
+        content: 'Total'
+      }
+    ];
+
+    const rankOptions = [
+      {
+        key: 'Adjust Rank',
+        text: 'Adjust Rank',
+        value: 'Adjust Rank',
+        content: 'Adjust Rank'
+      }, {
+        key: 'Hour',
+        text: 'Hour',
+        value: 'Hour',
+        content: 'Hour'
+      }, {
+        key: 'Night',
+        text: 'Night',
+        value: 'Night',
+        content: 'Night'
+      }
+    ];
+    return <div className="sorting-dropdowns" style={{
+        flex: 1,
+        flexDirection: 'row',
+        justifyContent: 'space-between',
+        display: 'inline-block'
+      }}>
+      <div style={{
+          display: 'inline-block',
+          paddingLeft: '10px'
+        }}>
+        <Icon name='calendar'/>
+        View by {' '}
+        <Dropdown labeled="labeled" inline="inline" options={lengthOptions} defaultValue={lengthOptions[0].value} onChange={(e, value) => this.handleChange(e, value.value)}/>
+        <Icon name='trophy'/>
+        Rank by {' '}
+        <Dropdown labeled="labeled" inline="inline" options={rankOptions} defaultValue={rankOptions[0].value} onChange={(e, value) => this.handleNightorHours(e, value.value)}/>
+      </div>
+    </div>
+
+  }
+
+  //begining of render function
+  render() {
```

Comment:
Please change the render function to the following:

```
render() {
    const { hours } = this.state;
    return (
      <Card fluid>
        <Card.Content>
          <Card.Header>Hour Breakdown</Card.Header>
        </Card.Content>
        <Card.Content>
          {this.dropDown()}
          <br/>
          <br/>
          <Card.Group>
            {  
              hours.map((user) => (this.customCard({ user, key: user.name })))
            }
          </Card.Group>
        </Card.Content>
      </Card>
    );
  }
}
```
I had typed a list of reasons why earlier, but I did not save my previous comment, so, just ask me in the meeting tonight and I will tell you. @vinitbytheminute

Reply:
Actually, I will type it out:
- You used class functions as components, `<this.dropDown />` and `<this.customCard />`. You cannot do that. In classes, functions stay functions, if you want to build a separate component, then build it outside of the class.
- Use Card.Header, it messes with the styling when you do otherwise
- You called a variable from the state, that you didn't even use. 
 
This is not everything I typed earlier, but just ask me when you get the chance and there are some conventions that I need to go over with you.